tfjs-backend-webgpu/src/device_limits.ts

```diff
@@ -15,12 +15,13 @@
  * =============================================================================
  */
 
-export const MAX_COMPUTE_WORKGROUP_INVOCATIONS = 256;
+export let maxComputeWorkgroupsPerDimension = 65535;
+export let maxComputeWorkgroupStorageSize = 16352;
+export let maxStorageBufferBindingSize = 134217728;
 
-export const MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE = 65535;
-
-export const MAX_COMPUTE_WORKGROUP_SIZE_X = 256;
-
-export const MAX_COMPUTE_WORKGROUP_SIZE_Y = 256;
-
-export const MAX_COMPUTE_WORKGROUP_SIZE_Z = 256;
+export function setDeviceLimits(deviceLimits: GPUSupportedLimits) {
+  maxComputeWorkgroupsPerDimension =
+      deviceLimits.maxComputeWorkgroupsPerDimension;
+  maxComputeWorkgroupStorageSize = deviceLimits.maxComputeWorkgroupStorageSize;
+  maxStorageBufferBindingSize = deviceLimits.maxStorageBufferBindingSize;
+}
```

Comment:
As mentioned above, I prefer to iterate a dict and initialize.

Reply:
> In addition, we can put them in a dict so that we can iterate them to initialize. The initial values here are not necessary.

Thanks for suggestion! I used adapter.limits to initialize a device, so these limits info could store in a `device` parameter of a backend instance, no need another file anymore, we could call tf.backend().device.limits where ever we want. PTAL.

> In index.ts, can we put all the creation of adapter and device into WebGPUBackend init?

Maybe not, a constructor is not possible to directly return a Promise, so async function `requestAdapter` and `requestDevice` cannot be called from a `new WebGPUBackend()`.